File: HelloWorld/Procedural1.cs
// Compile command: msc Procedural1.cs
// Execute commands:
//   mono Procedural1
//   mono Procedural1 "José Carlos Monteiro"
//   mono Procedural1 Universe

using System;

public class Procedural1
{

  public static void Main( string[] arguments )
  {
    string name = ( arguments.Length == 0 ) ? "World" : arguments[ 0 ];
    Console.WriteLine( "Hello, " + name + "!" );

    Console.WriteLine( "» José Carlos Monteiro «" );
  }

}

File: HelloWorld/Procedural1.java
// Compile command: javac -encoding "UTF-8" Procedural1.java
// Execute commands:
//   java -Dfile.encoding="UTF-8" Procedural1
//   java -Dfile.encoding="UTF-8" Procedural1 "José Carlos Monteiro"
//   java -Dfile.encoding="UTF-8" Procedural1 Universe

public class Procedural1 {

  public static void main( final String[] arguments ) {
    final String name = ( arguments.length == 0 ) ? "World" : arguments[ 0 ]; // Array.length is not procedural: it's a property of the object Array.
    System.out.printf( "Hello, %s!\n", name );

    System.out.println( "» José Carlos Monteiro «" );
  }

}